src/ska_dict.rs

```diff
@@ -97,10 +97,39 @@ where
 
     /// Iterates through all the k-mers from an input fastx file and adds them
     /// to the dictionary
-    fn add_file_kmers(&mut self, filename: &str, is_reads: bool, qual: &QualOpts) {
+    fn add_file_kmers(
+        &mut self,
+        filename: &str,
+        is_reads: bool,
+        qual: &QualOpts,
+        proportion_reads: Option<usize>,
+    ) {
+        // Going though the file once to count the number of reads
+        let mut reader =
+            parse_fastx_file(filename).unwrap_or_else(|_| panic!("Invalid path/file: {filename}"));
+
+        let mut nb_reads_total = 0;
+        while let Some(_record) = reader.next() {
+            nb_reads_total += 1;
+        }
+
+        let mut nb_reads = nb_reads_total;
+        if proportion_reads.is_some() {
+            nb_reads = (nb_reads as f32 * (proportion_reads.unwrap() as f32 / 100.0)) as usize;
+        }
+
+        let step = nb_reads_total as f64 / nb_reads as f64;
```

Comment:
Probably better to keep this as an integer step and avoid the f64 conversion  below (i.e. use round)